clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -328,6 +328,12 @@ public boolean poll(final long timeoutMs) {
                 // we need to ensure that the metadata is fresh before joining initially. This ensures
                 // that we have matched the pattern against the cluster's topics at least once before joining.
                 if (subscriptions.hasPatternSubscription()) {
+                    // Force metadata refresh, as long as the refresh backoff time has passed. This is to avoid a
+                    // situation where this consumer's metadata refresh happens after the current rebalance is done and
```

Comment:
Maybe we can improve the comment a bit more to clarify why we need to request metadata here. The important point is that, when a topic is created, the metadata update in any consumer can trigger the rebalance. In order to the topic creation to trigger one rebalance in the entire consumer group, we need each consumer to refresh metadata before it re-joins the group.

Currently the comment suggests, if we can refresh metadata before consumer re-joins the group, then consumer can avoid rebalance due to metadata refresh later. It seems a bit confusing.

Reply:
How about something like this:

For consumer group that uses pattern-based subscription, after a topic is created, any consumer that discovers the topic after metadata refresh can trigger rebalance across the entire consumer group. Multiple rebalance can be triggered after one topic creation if consumers refresh metadata at vastly different times. We can significantly reduce the number of rebalance caused by single topic creation by asking consumer to refresh metadata before re-joining the group as long as the refresh backoff time has passed.